pkg/services/user/userimpl/user.go

```diff
@@ -112,10 +145,82 @@ func (s *Service) Create(ctx context.Context, cmd *user.CreateUserCommand) (*use
 		}
 		_, err = s.orgService.InsertOrgUser(ctx, &orgUser)
 		if err != nil {
-			//  HERE ADD DELETE USER
+			err := s.store.Delete(ctx, userID)
 			return usr, err
 		}
 	}
 
 	return usr, nil
 }
+
+func (s *Service) Delete(ctx context.Context, cmd *user.DeleteUserCommand) error {
+	_, err := s.store.GetNotServiceAccount(ctx, cmd.UserID)
+	if err != nil {
+		return err
```

Comment:
nit
we could wrap the original error and differentiate the different error cases to easier debug failures.